### PR TITLE
[#95710624] Use GPG to encrypt the vault password

### PR DESCRIPTION
**Pull request by jimconner**
Distributing the ansible vault password throughout the team is a needless challenge which we can avoid by using multi-key encryption via GPG. The ansible-vault is still encrypted

**Makefile**
- Addition of a import-gpg-keys stanza to the Makefile which will import keys listed in gpg.recipients into the GPG keyring
- Addition of a recrypt stanza to the Makefile which will decrypt the vault, generate a new pass-phrase, PGP-encrypt the pass-phrase and then re-encrypt the vault

**ansible.cfg**
- A change to the ansible.cfg to tell ansible to use the open_the_vault.sh script to get the vault decryption password

**gpg.recipients**
- Addition of a gpg.recipients file which lists KEY-ID and email address for each approved recipient

**open_the_vault.sh**
- Addition of an open_the_vault.sh script which uses GPG to decrypt the vault pass-phrase and outpus it on stdout

**vault_passphrase.gpg**
- Addition of the GPG-encrypted vault passphrase

**vault**
- Re-encrypted with new passphrase for GPG recipients

**vault_password.txt**
- Deprecated - no longer needed - removed.

### CODE REVIEW

File: gpg.recipients [hunk 1]
```diff
@@ -0,0 +1,5 @@
+BCCB8A5F: "Anna Shipman <anna.shipman@digital.cabinet-office.gov.uk>"
+22652456: "Dan Carley (Government Digital Service) <dan.carley@digital.cabinet-office.gov.uk>"
+77AACBC2: "Jim Conner (I failed the turing test) <jim.conner@digital.cabinet-office.gov.uk>"
+B81D5E45: "Martin Jackson <martin@uncommonsense-uk.com>"
+F35E9C9E: "multicloud deploy <the-multi-cloud-paas-team+jenkins@digital.cabinet-office.gov.uk>"
```
Comment (dcarley) [inline gpg.recipients]: Does Jenkins need access to this vault?

File: Makefile [hunk 2]
```diff
@@ -15,7 +17,14 @@ vagrant-up:
...
 aws-provision: 
-	ansible-playbook -i localhost, aws-provision.yml -v --ask-vault-pass
+	ansible-playbook -i localhost, aws-provision.yml -v
 
 site: 
-	ansible-playbook -i ec2.py site.yml -v --ask-vault-pass
+	ansible-playbook -i ec2.py site.yml -v
+
+import-gpg-keys:
+	$(foreach var,$(shell cat gpg.recipients | awk -F: '{print $$1}'),gpg --keyserver hkp://keyserver.ubuntu.com --recv-keys $(var);)
```
Comment (dcarley) [inline Makefile]: Could you update this to match alphagov/tsuru-ansible@967c65e7621c36d0f4535a65a9ecd61a9dc36b43?

Might also be worth pulling in alphagov/tsuru-ansible#60, if that's going to get merged.

Comment (actionjack) [inline Makefile]: Can we merge this one in and then do a follow up PR to do the refactoring of the Makefile? It seems a shame to miss the thread of how we got here/there.

Comment (dcarley) [inline Makefile]: Yeah, for the refactoring (wrapping lines). But I don't think we should merge this with different behaviour (keys imported without prompt) than the other project.

**Comment by actionjack** [comment]
@dcarley changes made.

**Comment by dcarley** [comment]
Thanks :+1:

**Pull request merged**